src/sonic-yang-models/yang-models/sonic-buffer-queue.yang

```diff
@@ -54,8 +52,13 @@ module sonic-buffer-queue {
 
                 leaf profile {
                     default 0;
-                    type leafref {
-                        path "/bpf:sonic-buffer-profile/bpf:BUFFER_PROFILE/bpf:BUFFER_PROFILE_LIST/bpf:name";
+                    type union {
+                        type leafref {
+                            path "/bpf:sonic-buffer-profile/bpf:BUFFER_PROFILE/bpf:BUFFER_PROFILE_LIST/bpf:name";
+                        }
+                        type string {
+                            pattern "NULL";
```

Comment:
The dynamic calculation is supported in `BUFFER_PG` only.
For `BUFFER_QUEUE` table, the `profile` should always be defined in the `BUFFER_PROFILE` table. So no need to change `BUFFER_QUEUE`.

Reply:
@stephenxs  Are there other scenarios where NULL buffer profile may be configured apart from Dynamic calculation?

Reply:
Only two:
- `profile` in `BUFFER_PG` can be `NULL`, which is covered in this PR
- `size` in `BUFFER_POOL` is [optional](https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-buffer-pool.yang#L49), which has been taken into account